client/jetpack-onboarding/main.jsx

```diff
@@ -141,6 +142,16 @@ class JetpackOnboardingMain extends React.PureComponent {
 				<DocumentHead
 					title={ get( STEP_TITLES, stepName ) + ' ‹ ' + translate( 'Jetpack Start' ) }
 				/>
+				{ /* It is important to use `<QuerySites siteId={ siteSlug } />` here, however wrong that seems.
+				   * The reason is that we rely on an `isRequestingSite()` check to tell whether we've
+				   * finished fetching site details, which will tell us whether the site is connected,
+				   * which we need in turn to conditionally send JPO auth credentials (see below).
+				   * However, if we're logged out, we cannot `<QuerySites allSites />`,
+				   * since the concept of "all of a user's sites" doesn't make sense if there is no user.
+				   * We also cannot `<QuerySites siteId={ siteId } />` prior to having obtained the `siteId`
+				   * through JPO -- a Catch-22 situation.
+				   * Fortunately, querying by `siteSlug` works, since it's supported by underlying actions. */ }
+				<QuerySites siteId={ siteSlug } />
```

Comment:
We'll need to fix the `siteId` propType to accept a slug.

Reply:
D'oh, merged, but forgot to fix the propType and selector JSDoc. Will spin up a PR to do that...

Reply:
#23526